src/cpp/r/RSexp.cpp

```diff
@@ -838,6 +838,77 @@ bool inherits(SEXP object, const char* S3Class)
    return Rf_inherits(object, S3Class);
 }
 
+std::set<std::string> makeNseFunctions()
+{
+   std::set<std::string> nseFunctions;
+   nseFunctions.insert("quote");
+   nseFunctions.insert("substitute");
+   nseFunctions.insert("match.call");
+   nseFunctions.insert("library");
+   nseFunctions.insert("require");
+   nseFunctions.insert("enquote");
+   nseFunctions.insert("expression");
+   nseFunctions.insert("evalq");
+   nseFunctions.insert("subset");
+   return nseFunctions;
+}
+
+std::set<std::string> nseFunctions()
+{
+   static const std::set<std::string> set = makeNseFunctions();
+   return set;
+}
+
+bool maybePerformsNSEImpl(SEXP node)
+{
+   if (Rf_isExpression(node))
+   {
+      int n = Rf_length(node);
+      for (int i = 0; i < n; i++)
+         if (maybePerformsNSEImpl(VECTOR_ELT(node, i)))
+            return true;
+   }
+   
+   else if (Rf_isLanguage(node))
+   {
+      SEXP symbol = CAR(node);
+      if (Rf_isSymbol(symbol))
```

Comment:
Thanks, I noticed that as well, although the `is` functions for language objects seem to be well behaved. I'll make that change anyhow because I agree that's clearer.